base/src/matrix.cu

```diff
@@ -772,9 +772,11 @@ MatrixBase<T_Config>::setupMatrix(Solver<T_Config> *outer_solver, AMG_Config &cf
 
     if (m_separation_interior > m_separation_exterior) { FatalError("Interior separation cannot be wider than the exterior separation", AMGX_ERR_CONFIGURATION); }
 
-    int min_rows_latency_hiding = cfg.getParameter<int>("min_rows_latency_hiding", "default");
-
-    if (min_rows_latency_hiding < 0 || this->get_num_rows() < min_rows_latency_hiding) { m_separation_interior = m_separation_exterior; }
+    // If latency hiding is disabled, the interior is overwritten
+    if(!isLatencyHidingEnabled(cfg))
+    {
+        m_separation_interior = m_separation_exterior;
```

Comment:
this is the condition we check in the solvers to see if latency hiding could be enabled, right?

Reply:
Yes - it was previously just checking whether the local num rows was less than the threshold. The problem is that if you have an unequal number of rows per rank that can lead to different ranks having different views of where latency hiding is enabled / disabled and that causes deadlocks. This patch fixes that by adding a collective check.

There are some minor optimisations that could be applied but I'm trying to get this fixed ASAP for a customer release.